lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      paths = Set.new(applicable_files.map { |file| File.dirname(file) + File::SEPARATOR })
+      repo_files = Set.new(Overcommit::GitRepo.list_files(paths.to_a) + applicable_files)
```

Comment:
Yup. Converting to a `Set` may be too much overhead for large commits though, considering `git ls-tree` ignores duplicate paths anyway. Likely negligible compared to linter run times, though. What do you think?